ops.qmd

```diff
@@ -2,84 +2,760 @@
+
+The data engineering team builds and sustains the data foundation for reliable model development and operations. Their work enables data scientists and ML engineers to focus on building, training and deploying ML models at scale.
+
+### Data Scientists
+
+The job of the data scientists is to focus on the research, experimentation, development and continuous improvement of ML models. They leverage their expertise in statistics, modeling and algorithms to create high-performing models. Their main responsibilities include:
+
+* Working with business and data teams to identify opportunities where ML can add value. Framing the problem and defining success metrics.
+* Performing exploratory data analysis to understand relationships in data and derive insights. Identifying relevant features for modeling.
+* Researching and experimenting with different ML algorithms and model architectures based on the problem and data characteristics. Leveraging libraries like TensorFlow, PyTorch, Keras.
+* Training and fine-tuning models by tuning hyperparameters, adjusting neural network architectures, feature engineering, etc. to maximize performance.
+* Evaluating model performance through metrics like accuracy, AUC, F1 scores. Performing error analysis to identify areas for improvement.
+* Developing new model versions by incorporating new data, testing different approaches, and optimizing model behavior. Maintaining documentation and lineage for models.
+
+For example, a data scientist may leverage TensorFlow and TensorFlow Probability to develop a demand forecasting model for retail inventory planning. They would iterate on different sequence models like LSTMs and experiment with features derived from product, sales and seasonal data. The model would be evaluated based on error metrics versus actual demand before deployment. The data scientist monitors performance and retrains/enhances the model as new data comes in.
+
+Data scientists drive model creation, improvement and innovation through their expertise in ML techniques. They collaborate closely with other roles to ensure models create maximum business impact.
+
+### ML Engineers
+
+ML engineers enable models data scientists develop to be productized and deployed at scale. Their expertise makes models reliably serve predictions in applications and business processes. Their main responsibilities include:
+
+* Taking prototype models from data scientists and hardening them for production environments through coding best practices.
+* Building APIs and microservices for model deployment using tools like Flask, FastAPI. Containerizing models with Docker.
```

Comment:
Possible links for Flask and FastAPI:
- https://flask.palletsprojects.com/en/3.0.x/
- https://fastapi.tiangolo.com/

Reply:
🙏